src/lib.rs

```diff
@@ -339,7 +381,9 @@ impl AstarteSdk {
     }
 
     async fn send_introspection(&self) -> Result<(), AstarteError> {
-        let introspection = self.interfaces.get_introspection_string();
+        let interfaces_arc = self.interfaces.clone();
```

Comment:
Call this `interfaces_mutex` as all other cases where you cloned the `self.interfaces`.